### PR TITLE
Part1. RxJava2 support

### PR DESCRIPTION
**Pull request by nikitin-da**
Sorry guys for such huge amount of code, but I have no idea how to make it smaller =(
The both parts (SQLite and ContentResolver) and tests are highly dependent on common api and I'm not sure how interop library can help us.

This part do not handle case when there is no data in storage, I will solve it in next release (proposals about naming of our optional are welcome ;) )

Part of #685

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -53,56 +53,60 @@ def daggerVersion = '2.4'
...
         // Core libraries
-        supportAnnotations             : 'com.android.support:support-annotations:' + supportLibsVersion,
-        rxJava                         : 'io.reactivex:rxjava:1.2.1',
-        kotlinStdLib                   : "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version",
+        supportAnnotations                       : 'com.android.support:support-annotations:' + supportLibsVersion,
+
+        // TODO remove RxJava1 and Interop once RxJava2 migration will be complete.
+        rxJava                                   : 'io.reactivex:rxjava:1.2.1',
+        rxJavaInterop                            : 'com.github.akarnokd:rxjava2-interop:0.10.0',
+        rxJava2                                  : 'io.reactivex.rxjava2:rxjava:2.1.4',
```
Comment (artem-zinnatullin) [inline build.gradle]: 2.1.6

File: build.gradle [hunk 2]
```diff
@@ -53,56 +53,60 @@ def daggerVersion = '2.4'
...
 ext.libraries = [
         // Core libraries
-        supportAnnotations             : 'com.android.support:support-annotations:' + supportLibsVersion,
-        rxJava                         : 'io.reactivex:rxjava:1.2.1',
-        kotlinStdLib                   : "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version",
+        supportAnnotations                       : 'com.android.support:support-annotations:' + supportLibsVersion,
+
+        // TODO remove RxJava1 and Interop once RxJava2 migration will be complete.
+        rxJava                                   : 'io.reactivex:rxjava:1.2.1',
+        rxJavaInterop                            : 'com.github.akarnokd:rxjava2-interop:0.10.0',
```
Comment (artem-zinnatullin) [inline build.gradle]: 0.10.9

**Comment by artem-zinnatullin** [comment]
@geralt-encore let's merge this?

**Comment by geralt-encore** [comment]
Sorry, just saw this PR after your mention.
I'll review it today, I promise! :)

**Comment by geralt-encore** [comment]
I think that this coverage drop is not that important, so I'll merge it right away

**Pull request merged**